### PR TITLE
#18407 this fix allows you to save not only file asset on the edit co…

### PR DESCRIPTION
**Pull request by jdotcms**
this fix allows you to save not only file asset on the edit contentlet but also, dotAsset and any contentlet with binary field

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/ajax/FileAssetAjax.java [hunk 1]
```diff
@@ -56,41 +58,50 @@ public FileAssetAjax() {
...
+
 			os.write(newText.getBytes());
 		} catch(Exception e) {
-			Logger.error(getClass(), "Error writing to file", e);
-		}finally {
-			if (os != null){
-				os.close();
-			}
 
+			Logger.error(getClass(), "Error writing to file", e);
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/ajax/FileAssetAjax.java:103]: should this logic get into a API?

File: dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/ajax/FileAssetAjax.java [hunk 2]
```diff
@@ -56,41 +58,50 @@ public FileAssetAjax() {
...
+
+		final WebContext webContext        = WebContextFactory.get();
+		final HttpServletRequest request   = webContext.getHttpServletRequest();
+		final User user 			       = userAPI.getLoggedInUser(request);
+		final boolean respectFrontendRoles = userAPI.isLoggedToFrontend(request);
+		final Contentlet contentlet        = APILocator.getContentletAPI().find(contentletInode, user, respectFrontendRoles);
+		String incomingFileName            = null;
+		String binaryFieldName             = binField;
+
+		if (contentlet.isFileAsset()) {
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/ajax/FileAssetAjax.java:72]: I think we should support all binary fields - even if you just have a generic content that has a binary field with text

Comment (jdotcms) [reply dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/ajax/FileAssetAjax.java:72]: last if support this case, I have tested with a contentlet with a binary as a field

File: dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/ajax/FileAssetAjax.java (hunk 1 above)
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/ajax/FileAssetAjax.java:103]: I would probably promote to an endpoint, I might create an issue

Comment (jdotcms) [reply dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/ajax/FileAssetAjax.java:103]: https://github.com/dotCMS/core/issues/18421

**Review by erickgonzalez — APPROVED** [review]

**Review by freddyucv — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

**Review by wezell — APPROVED** [review]

**Pull request merged**